src/server/license/server/api_server.go

```diff
@@ -29,17 +32,76 @@ type apiServer struct {
 	license col.PostgresCollection
 }
 
-// New returns an implementation of license.APIServer.
-func New(env Env) (lc.APIServer, error) {
+// New returns an implementation of license.APIServer, and a function that bootstraps the license server via environment.
+func New(env Env) (lc.APIServer, func(context.Context) error, error) {
 	s := &apiServer{
 		env:     env,
 		license: licenseCollection(env.DB, env.Listener),
 	}
-	return s, nil
+	return s, s.envBootstrap, nil
+}
+
+func (a *apiServer) envBootstrap(ctx context.Context) error {
+	if a.env.Config.LicenseKey == "" && a.env.Config.EnterpriseSecret == "" {
+		return nil
+	}
+	if a.env.Config.LicenseKey == "" || a.env.Config.EnterpriseSecret == "" {
+		return errors.New("License server failed to bootstrap via environment; Either both or neither of LICENSE_KEY and ENTERPRISE_SECRET must be set.")
+	}
+	logrus.Info("Started to configure license server via environment")
+	localhostPeerAddr := "grpc://localhost:" + fmt.Sprint(a.env.Config.PeerPort)
+	if err := func() error {
+		ctx = auth.AsInternalUser(ctx, "license-server")
+		_, err := a.activate(ctx, &lc.ActivateRequest{ActivationCode: a.env.Config.LicenseKey})
+		if err != nil {
+			return errors.Wrapf(err, "failed to activate the license service")
+		}
+		_, err = a.AddCluster(ctx, &lc.AddClusterRequest{
+			Id:               localhostEnterpriseClusterId,
+			Address:          localhostPeerAddr,
+			UserAddress:      localhostPeerAddr,
+			Secret:           a.env.Config.EnterpriseSecret,
+			EnterpriseServer: true,
+		})
+		if err != nil {
+			if errors.As(err, lc.ErrDuplicateClusterID) {
+				_, err = a.UpdateCluster(ctx, &lc.UpdateClusterRequest{
+					Id:          localhostEnterpriseClusterId,
+					Address:     localhostPeerAddr,
+					UserAddress: localhostPeerAddr,
+					Secret:      a.env.Config.EnterpriseSecret,
+				})
+				if err != nil {
+					return errors.Wrapf(err, "failed to update localhost cluster in the license service")
```

Comment:
One last nitpicky comment.  I think it's fine to omit the "failed" part from error constructors.  If it succeeded, it wouldn't be an Error object.  So this could be errors.Wrapf(err, "update localhost cluster").  (Ideally whatever handles the returned error would add "bootstrap license service".)